app/src/main/java/com/vero/tasky/agenda/presentation/agenda/AgendaScreen.kt

```diff
@@ -29,9 +30,8 @@ import kotlinx.coroutines.launch
 @Composable
 fun AgendaScreen(
     viewModel: AgendaViewModel = hiltViewModel(),
-    openEventScreen: (isEditable: Boolean) -> Unit,
-    openTaskScreen: (isEditable: Boolean) -> Unit,
-    openReminderScreen: (isEditable: Boolean) -> Unit
+    openAgendaItemScreen: (agendaItem: AgendaItem, isEditable: Boolean) -> Unit,
+    openNewAgendaItemScreen: (agendaItemType: AgendaItemType) -> Unit,
```

Comment:
I would rename these to
onAgendaItemClick
onNewAgendaItemClick

Mostly cause the AgendaScreen shouldn't know what should happen when the item is clicked (In this case, it shouldn't know it will open a screen)

It's optional tho. In this case, nothing will change. 
But in a future app, you might have the parameter "openNewScreen". Which at first, might open a new screen. But later gets refactored to do something different, then you'd also need to remember to rename the parameter